agent/www/shared/address_service.js

```diff
@@ -250,19 +301,13 @@ AddressService.prototype.update_user = function (c) {
 
 AddressService.prototype.on_message = function (context) {
     if (context.message.subject === 'address') {
-        this.update(context.message.body);
+        this.update_address(context.message.body);
```

Comment:
I think I'd be inclined to inline update_address.  update_address is now quite short and it would mean you can clearly see on the callback emitted by on_message in one place.